web/api/v1/api.go

```diff
@@ -372,6 +372,24 @@ func (api *API) query(r *http.Request) (result apiFuncResult) {
 		return apiFuncResult{nil, returnAPIError(res.Err), res.Warnings, qry.Close}
 	}
 
+	if res.Value.Type() == "vector" {
```

Comment:
It's usually better to use some `const` than hardcoded strings. Maybe we can pull it out from some package? (: (e.g `promql` or `model`